src/presenter/films-presenter.js

```diff
@@ -105,37 +153,41 @@ export default class FilmsPresenter {
     if (this.#currentSortType === sortType) {
       return;
     }
-    // - Сортируем задачи
-    this.#sortFilms(sortType);
-    // - Очищаем список
-    this.#clearFilmList();
-    // - Рендерим список заново
-    this.#renderFilmsList();
+
+    this.#currentSortType = sortType;
+    this.#clearFilm({resetRenderedMovieCount: true});
+
+    this.#renderMovie();
   };
 
   #renderSort = () => {
-    render(this.#sortComponent, this.#filmListContainer);
+    this.#sortComponent = new SortView(this.#currentSortType);
     this.#sortComponent.setSortTypeChangeHandler(this.#handleSortTypeChange);
+
+    render(this.#sortComponent, this.#filmListContainer);
   };
 
   #renderFilmsList = () => {
     render(this.#filmSection, this.#filmListContainer);
-    render(this.#filmContainer, this.#filmSection.element);
-
-    this.#renderFilms(0, Math.min(this.#sectionMovie.length, SHOW_FILM_COUNT_STEP));
-
-    if (this.#sectionMovie.length > SHOW_FILM_COUNT_STEP) {
-      this.#renderLoadMoreButton();
-    }
   };
 
   #renderMovie = () => {
-    if (this.#sectionMovie.length === 0) {
+    const movies = this.movies;
+    const movieCount = movies.length;
+
+    if (movieCount === 0) {
       this.#renderNoFilms();
       return;
     }
 
     this.#renderSort();
+    render(this.#filmContainer, this.#filmSection.element);
     this.#renderFilmsList();
+
```

Comment:
Добавим вызов метода для обновления модального окна
```suggestion
this.#updateOpenedModal();
```